### PR TITLE
source-gmail contribution from topefolorunso

### LINKED ISSUE #46654: 🎉 New Source Gmail
### Discussed in https://github.com/airbytehq/airbyte/discussions/36115

<div type='discussions-op-text'>

<sup>Originally posted by **elattma** February 17, 2023</sup>
## Tell us about the new connector you’d like to have
* Which source and which destination?
* Source gmail
* Do you need a specific version of the underlying data source e.g: you specifically need support for an older version of the API or DB? 
* No

## Describe the context around this new connector
* Why do you need this integration? How does your team intend to use the data? This helps us understand the use case.
* To read data from gmail. We intend to index the text content for later use.
* How often do you want to run syncs?
* Daily or weekly I think is fine.
* If this is an API source connector, which entities/endpoints do you need supported?
* N/A
* If the connector is for a paid service, can we name you as a mutual user when we subscribe for an account? Which company should we name? 
* Yes

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?
Building out the data pipeline myself.

## Are you willing to submit a PR?
Yes!! I'm more than happy to :)
</div>

### PR DESCRIPTION
**Pull request by topefolorunso**
## What

This PR adds a new connector for source Gmail (source-gmail).

Contributor-provided connector description:

Gmail is the email service provided by Google.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | OAuth Client ID.  |  |
| `client_secret` | `string` | OAuth Client Secret.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `includeSpamTrash` | `boolean` | Include Spam &amp; Trash. Include drafts/messages from SPAM and TRASH in the results. Defaults to false. | false |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| profile |  | No pagination | ✅ |  ❌  |
| drafts | id | DefaultPaginator | ✅ |  ❌  |
| labels | id | No pagination | ✅ |  ❌  |
| labels_details | id | No pagination | ✅ |  ❌  |
| messages | id | DefaultPaginator | ✅ |  ❌  |
| messages_details | id | No pagination | ✅ |  ❌  |
| threads | id | DefaultPaginator | ✅ |  ❌  |
| threads_details | id | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/gmail.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Gmail is the email service provided by Google.
+
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `client_id` | `string` | OAuth Client ID.  |  |
+| `client_secret` | `string` | OAuth Client Secret.  |  |
+| `client_refresh_token` | `string` | Refresh token.  |  |
+| `includeSpamTrash` | `boolean` | Include Spam &amp; Trash. Include drafts/messages from SPAM and TRASH in the results. Defaults to false. | false |
```
Comment (natikgadzhi) [inline docs/integrations/sources/gmail.md:11]: @topefolorunso can you re-upload and snake_case_the_config_input_name here? I.e. `include_spam_and_trash`.

Comment (natikgadzhi) [reply docs/integrations/sources/gmail.md:11]: Ah, you know what, I think I will search and replace it here instead. One sec.

Comment (natikgadzhi) [inline docs/integrations/sources/gmail.md:11]:
```suggestion
| `include_spam_and_trash` | `boolean` | Include Spam &amp; Trash. Include drafts/messages from SPAM and TRASH in the results. Defaults to false. | false |
```

**Review by natikgadzhi — APPROVED** [review]

File: docs/integrations/sources/gmail.md (hunk 1 above)
Comment (topefolorunso) [reply docs/integrations/sources/gmail.md]: i'll need to change that in manifest too. one sec

**Pull request merged**